src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,38 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String paragraph = System.lineSeparator();
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ")
+                .append(dateTo);
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("d.MM.yyyy");
+        LocalDate startDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate endDate = LocalDate.parse(dateTo, formatter);
+
+        for (String name: names) {
+            int totalSalary = 0;
+
+            for (String employeeData: data) {
+                String[] employeeInfo = employeeData.split(" ");
+                LocalDate workDay = LocalDate.parse(employeeInfo[0], formatter);
+
+                if (name.equals(employeeInfo[1])
+                        && workDay.isAfter(startDate)
+                        && (workDay.isEqual(endDate)
+                        || workDay.isBefore(endDate))) {
```

Comment:
I believe we can simplify this if statement by only checking on `!isBefore` and `!isAfter`, what do you think?